gprofiler/profilers/perf.py

```diff
@@ -245,6 +246,14 @@ def __init__(
         self._node_processes_attached: List[Process] = []
         self._perf_memory_restart = perf_memory_restart
         switch_timeout_s = duration * 3  # allow gprofiler to be delayed up to 3 intervals before timing out.
+        extra_args = []
+        if perf_mode in ("fp", "dwarf", "smart"):
```

Comment:
This check is actually "is perf enabled". Nicer to check as `if not disabled` and I think that we already have the `not in ("none", "disabled")` check somewhere - please share.

Reply:
gotcha, please note the `cast` - I think it's safe but anyways I can just accept `int | bool | str | None` (`user_args.get` ret type) in `is_profiler_disabled`
f370b148ff1e80657f395d4044fa25fe530cd008

Reply:
Cast is ok